File: kruskal.py
def find(parent, x):
    if parent[x] != x:
        parent[x] = find(parent, parent[x])
    return parent[x]

def union(parent, rank, u, v):
    u_root = find(parent, u)
    v_root = find(parent, v)
    
    if rank[u_root] > rank[v_root]:
        parent[v_root] = u_root
    elif rank[u_root] < rank[v_root]:
        parent[u_root] = v_root
    else:
        parent[v_root] = u_root
        rank[u_root] += 1

def kruskal(n, edges):
    edges.sort(key=lambda x: x[2])
    parent = list(range(n))
    rank = [0] * n
    
    total_weight = 0
    selected_edges = []
    for u, v, weight in edges:
        u_root = find(parent, u)
        v_root = find(parent, v)
        
        if u_root != v_root:
            selected_edges.append((u, v, weight))
            total_weight += weight
            union(parent, rank, u_root, v_root)
            
    return total_weight, selected_edges

if __name__ == "__main__":
    n = int(input("Enter the number of vertices: "))
    m = int(input("Enter the number of edges: "))
    
    edges = []
    print("Enter the edges (u, v, weight):")
    for _ in range(m):
        u, v, weight = map(int, input().split())
        edges.append((u, v, weight))
    
    total_weight, selected_edges = kruskal(n, edges)
    
    print("Selected edges:")
    for edge in selected_edges:
        print("({}, {}, {})".format(*edge))
    print("Total weight of the minimum spanning tree:", total_weight)

File: kruskal.cpp
#include <algorithm>
#include <iostream>
#include <vector>
#include <tuple>

using namespace std;

struct Edge {
    int u, v, weight;
    
    bool operator<(const Edge &other) const {
        return weight < other.weight;
    }
};

int find(vector<int> &parent, int x) {
    if (parent[x] != x) {
        parent[x] = find(parent, parent[x]);
    }
    return parent[x];
}

void kruskal(int n, vector<Edge> &edges) {
    sort(edges.begin(), edges.end());
    vector<int> parent(n);
    for (int i = 0; i < n; i++) {
        parent[i] = i;
    }
    
    int totalWeight = 0;
    for (Edge &edge : edges) {
        int uRoot = find(parent, edge.u);
        int vRoot = find(parent, edge.v);
        
        if (uRoot != vRoot) {
            cout << "Selected edge: (" << edge.u << ", " << edge.v << ", " << edge.weight << ")\n";
            totalWeight += edge.weight;
            parent[uRoot] = vRoot;
        }
    }
    cout << "Total weight of the minimum spanning tree: " << totalWeight << "\n";
}

int main() {
    int n, m;
    cout << "Enter the number of vertices and edges: ";
    cin >> n >> m;
    
    vector<Edge> edges(m);
    cout << "Enter the edges (u, v, weight):" << endl;
    for (int i = 0; i < m; i++) {
        cin >> edges[i].u >> edges[i].v >> edges[i].weight;
    }
    
    kruskal(n, edges);
    return 0;
}